src/bandersnatch/utils.py

```diff
@@ -139,3 +139,37 @@ def bandersnatch_safe_name(name: str) -> str:
     bandersnatch also lower cases the returned name
     """
     return SAFE_NAME_REGEX.sub("-", name).lower()
+
+
+# From https://peps.python.org/pep-0616/
+def removeprefix(original: str, prefix: str) -> str:
+    if original.startswith(prefix):
+        prefix_len = len(prefix)
+        mod_str = original[prefix_len:]
+        return mod_str
+    else:
+        return original
+
+
+# Python tags https://peps.python.org/pep-0425/#python-tag
+def parse_version(version: str) -> List[str]:
+    _versions: List[str] = []
+
+    _version_with_dot = removeprefix(version.lower(), "py")
+    _version_without_dot = _version_with_dot.replace(".", "")
+
+    pep425_pyver_str_cp = f"-cp{_version_without_dot}-"
+    pep425_pyver_str_ip = f"-ip{_version_without_dot}-"
+    pep425_pyver_str_jy = f"-jy{_version_without_dot}-"
+    pep425_pyver_str_pp = f"-pp{_version_without_dot}-"
+    pep425_pyver_str_py1 = f"-py{_version_with_dot}-"
+    pep425_pyver_str_py2 = f"-py{_version_with_dot}."
+
+    _versions.extend([pep425_pyver_str_cp])
+    _versions.extend([pep425_pyver_str_ip])
+    _versions.extend([pep425_pyver_str_jy])
+    _versions.extend([pep425_pyver_str_pp])
+    _versions.extend([pep425_pyver_str_py1])
+    _versions.extend([pep425_pyver_str_py2])
```

Comment:
Does this logic exist in the Python packaging or another core library? Just want to avoid maintaining it if it's in a core library.
- My quick look at [packaging](https://github.com/pypa/packaging) says no ...

Reply:
There is a dictionary of interpreters and their short names which would allow to not hard code a list at least: https://packaging.pypa.io/en/latest/tags.html#packaging.tags.INTERPRETER_SHORT_NAMES

Reply:
Here's my attempt at incorporating the comments - would you check if this would suffice?
I found that two types of separators are needed for an edge case: [Numpy 1.7.1](https://pypi.org/project/numpy/1.7.1/#files)

```python
from packaging.tags import INTERPRETER_SHORT_NAMES
from typing import Dict, List


def parse_version(version: str) -> List[str]:
    _versions: List[str] = []
    
    interpreters = INTERPRETER_SHORT_NAMES.values()
    tag_separator1 = '-'
    tag_separator2 = '.'
    
    _versions.extend([tag_separator1 + i + version + tag_separator1 for i in interpreters])
    _versions.extend([tag_separator1 + 'py' + version + tag_separator2])
    
    return _versions


parse_version('3.7')
# ['-py3.7-', '-cp3.7-', '-pp3.7-', '-ip3.7-', '-jy3.7-', '-py3.7.']

```

![image](https://user-images.githubusercontent.com/2081333/162878213-d55aeeeb-aede-45cc-8b7c-e9cd697bc4aa.png)